### ISSUE TITLE
Rectangle opacity after $img->fit

### ISSUE THREAD

**Issue by FragsterAt** [Intervention/image#994]
I have strange opacity bug after appling fit() function to the image:

```php
$img = Image::make(Storage::get($savePath . 'background_original.jpg'));
$img->fit($puzzleImage->width, $puzzleImage->height);
		$img->rectangle($x - $s, $y - $s, $x + $s, $y + $s, function ($draw) {
			$draw->background('rgba(0, 0, 0, 0.5)');
		});
Storage::put($savePath . "/background_resized.jpg", (string) $img->encode('jpg', 92));
//$img = Image::make(Storage::get($savePath . 'background_resized.jpg'));
$img->rectangle(100, 100, 200, 200, function ($draw) {
	$draw->background('rgba(0, 0, 0, 0.5)');
});
Storage::put($savePath . "/background.jpg", (string) $img->encode('jpg', 92));
```
in this code I have black square instead of translucent. But if i uncomment ```//$img = Image::make(Storage::get($savePath . 'background_resized.jpg'));``` (or comment $img->fit ) the result is ok.
Also it begin work fine if I draw text **with ttf font** before drawing rectangle. May be in some other cases too.

**Comment by FragsterAt** [comment]
Using php 7.27 with GD bundled (2.1.0 compatible)

**Comment by westy92** [comment]
I'm having this issue as well. PHP 7.4, linux, intervention/image version: 2.5.1.

**Comment by westy92** [comment]
I found a temporary fix. If you create a new canvas with a transparent background color then insert it to your existing image, it works.

```php
$img->fit($puzzleImage->width, $puzzleImage->height);
$overlay = Image::canvas(100, 100, 'rgba(0, 0, 0, 0.5)');
$img->insert($overlay, 'bottom');
```

**Comment by olivervogel** [comment]
This issue has been resolved as of version `3.2`.
